### ISSUE TITLE
Single precision grid files

### ISSUE THREAD

**Issue by abyrd** [conveyal/r5#52]
Currently we're using double-precision floats to store opportunity density. The files shipped over the network are zipped and reasonably small (a few hundred kB), but the grids can take up hundreds of MB when uncompressed. This is not excessive for a modern web app, but still not necessary and contributes to high resource consumption.

Single-precision floating point numbers are sufficiently accurate and would cut memory consumption in half. Single-precision can represent all integers with six or less significant digits, and can store values up to the 10^38 range, which is probably sufficient to count the individual atoms in our analysis grid cells.

**Comment by mattwigway** [comment]
At zoom level 9, which is what we're using, there are 2^(9 * 2) * 256 * 256 grid cells on Earth. There are [approximately 10^49](http://www.fnal.gov/pub/science/inquiring/questions/atoms.html) atoms on earth, meaning that there are 10^49 / (2^(9*2) * 256 * 256) = 10^49 / (1.7 * 10^10) underneath each cell (not including the atmosphere), putting us well in int overflow range if trying to measure accessibility to atoms :)

**Comment by abyrd** [comment]
Isn't that the estimated number of atoms in the full 3D volume of the earth? I was just guessing based on how many you could line up on the surface. I could be off by a lot, I was just trying to work out whether we were using excessive precision.

**Comment by mattwigway** [comment]
Yes, that's the full volume of the earth. Since we're unlikely to measure accessibility to atoms at all, I think the point is moot :)
